### ISSUE TITLE
Handle ENODEV in tuntap tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6931]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)| @mithrandi reported|
|-|-|
|Trac ID|trac#6931|
|Type|defect|
|Created|2014-01-27 13:11:43Z|
|Branch|https://github.com/twisted/twisted/tree/tuntap-enodev-6931|

If the tun kernel module is not loaded and not able to be loaded, then /dev/net/tun may exist but respond with ENODEV. The tests (`twisted.pair.test.test_tuntap.RealDevice*`) handle ENOENT, but not ENODEV, resulting in test failures like this:

```
[ERROR]
Traceback (most recent call last):
  File "/home/buildslave/slaves/twisted/bot-idnar-debian64/easy-no-zope-2.6debian/venv/lib/python2.6/site-packages/Twisted-13.2.0-py2.6-linux-x86_64.egg/twisted/pair/test/test_tuntap.py", line 342, in setUp
    os.O_RDWR | os.O_NONBLOCK)
  File "/home/buildslave/slaves/twisted/bot-idnar-debian64/easy-no-zope-2.6debian/venv/lib/python2.6/site-packages/Twisted-13.2.0-py2.6-linux-x86_64.egg/twisted/pair/test/test_tuntap.py", line 581, in open
    return super(TestRealSystem, self).open(filename, *args, **kwargs)
exceptions.OSError: [Errno 19] No such device: '/dev/net/tun'
```

<details><summary>Searchable metadata</summary>

```
trac-id__6931 6931
type__defect defect
reporter__mithrandi mithrandi
priority__normal normal
milestone__ 
branch__branches_tuntap_enodev_6931 branches/tuntap-enodev-6931
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__pair pair
keywords__ 
time__1390828303000000 1390828303000000
changetime__1391357352000000 1391357352000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [41518]) Branching to 'tuntap-enodev-6931'

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_3"></a>|@hynek set owner to @exarkun|
|-|-|

The bots are green and the code makes sense. I’d personally prefer a regression test but if you think it’s not necessary, feel free to merge.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_4"></a>|@hynek commented|
|-|-|

Ugh, never mind the tests comment. m(  That was a knee-jerk. :) Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

> Ugh, never mind the tests comment. m( That was a knee-jerk. :) Please merge.

He he. :)  Thanks for the review!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [41524]) Merge tuntap-enodev-6931

Author: exarkun
Reviewer: hynek
Fixes: [#6931](https://github.com/twisted/twisted/issues/6931)

Add skip logic to the tuntap test suite to handle the `ENODEV` error case
when trying to open a real tuntap device on the system.
